### PR TITLE
Clean and divers fix

### PR DESCRIPTION
**Pull request by utas-tajona**
- adding reset state when page component died
-  minor fix on job entity
-   fix file with empty space
-  adding batch insert for saveAll()

### CODE REVIEW

File: api/src/main/resources/application.properties [hunk 1]
```diff
@@ -50,3 +50,7 @@ app.api.version=@project.version@
 app.cors.max_age_secs=3600
 spring.servlet.multipart.max-file-size=600MB
 spring.servlet.multipart.max-request-size =600MB
+spring.jpa.properties.hibernate.generate_statistics=true
+spring.jpa.properties.hibernate.jdbc.batch_size=200
+spring.jpa.properties.hibernate.order_inserts=true
+hibernate.cache.use_query_cache=true
```
Comment (nspool) [inline api/src/main/resources/application.properties:56]: What are these changes related to?

File: ui/src/index.css [hunk 2]
```diff
@@ -46,4 +46,12 @@ header a:hover {
...
     height: 18px;
     line-height: 18px;
+}
+
+.MuiBackdrop-root {
+    z-index: 200 !important;
+}
+
+.MuiDialog-paper {
+    z-index: 201 !important;;
```
Comment (nspool) [inline ui/src/index.css:56]: Better not to use the `!important` directive or specify material ui css directly. Is this something that could be done in themes instead?

File: ui/src/components/store.js [hunk 3]
```diff
@@ -19,9 +19,9 @@ import { jobReducer } from './job/jobReducer';
 import jobWatcher from './job/jobMiddleware';
 
 const initialiseSagaMiddleware = createSagaMiddleware();
+const isDev = (process.env.NODE_ENV == 'development');
 
-const middleware = [
-  ...getDefaultMiddleware(),
+const middleware = (isDev) ? [...getDefaultMiddleware(), initialiseSagaMiddleware] : [
```
Comment (nspool) [inline ui/src/components/store.js:24]: What is this change for?

**Review by nspool — COMMENTED** [review]
Looks good. Can you pelase check that the Prettier style is being applied to the UI JS files?

File: api/src/main/resources/application.properties (hunk 1 above)
Comment (utas-tajona) [reply api/src/main/resources/application.properties:56]: yes this will increase speed at upload and not failed on the edge env

File: ui/src/index.css (hunk 2 above)
Comment (utas-tajona) [reply ui/src/index.css:56]: it's litteray the same rule usine z-index being override so it's above ag-grid

File: ui/src/components/store.js (hunk 3 above)
Comment (utas-tajona) [reply ui/src/components/store.js:24]: getDefaultMiddleware check lots of dev stuff like is the state immutable.
it slow down considerably the UI. 

Good for dev bad for the rest

File: api/src/main/resources/application.properties [hunk 4]
```diff
@@ -50,3 +50,7 @@ app.api.version=@project.version@
 app.cors.max_age_secs=3600
 spring.servlet.multipart.max-file-size=600MB
 spring.servlet.multipart.max-request-size =600MB
+spring.jpa.properties.hibernate.generate_statistics=true
+spring.jpa.properties.hibernate.jdbc.batch_size=200
```
Comment (jonescc) [inline api/src/main/resources/application.properties:54]: You could increase this - 200 is actually a low number - we use 1000 for our harvesters

**Review by jonescc — COMMENTED** [review]
Backend changes LGTM.   @nspool?

**Comment by nspool** [comment]
Looks good, but we should try and avoid using `!important` in future CSS. Can you please check that the Prettier style is being applied to the UI JS files? It should be using the prettierrc in the UI directory.

**Review by nspool — COMMENTED** [review]
Just needs formatting and merge conflicts resolved and should be good to merge.

**Review by nspool — APPROVED** [review]
